ipn/ipnlocal/serve.go

```diff
@@ -812,6 +812,24 @@ func (b *LocalBackend) getServeHandler(r *http.Request) (_ ipn.HTTPHandlerView,
 // we serve requests for. `backend` is a HTTPHandler.Proxy string (url, hostport or just port).
 func (b *LocalBackend) proxyHandlerForBackend(backend string) (http.Handler, error) {
 	targetURL, insecure := expandProxyArg(backend)
+
+	// Handle unix: scheme specially
+	if strings.HasPrefix(targetURL, "unix:") {
+		socketPath := strings.TrimPrefix(targetURL, "unix:")
```

Comment:
I think we should explicitly block the [Tailscale socket](https://github.com/tailscale/tailscale/blob/b7f0c9ccbf46a8180305d5b3bbcfdd0ff98c0094/cmd/tailscaled/tailscaled.go#L508) path here. Otherwise `tailscale serve` can be used to open up the socket to your local tailnet with full access (because connections are coming from tailscaled running as `root`).